collect_app/src/main/java/org/odk/collect/android/utilities/ExternalAppIntentProvider.java

```diff
@@ -17,7 +17,16 @@ public class ExternalAppIntentProvider {
     private static final String URI_KEY = "uri_data";
 
     public Intent getIntentToRunExternalApp(FormController formController, FormEntryPrompt formEntryPrompt) throws ExternalParamsException, XPathSyntaxException {
-        String exSpec = formEntryPrompt.getAppearanceHint().replaceFirst("^ex[:]", "");
+        String appearance = formEntryPrompt.getAppearanceHint();
+
+        String exSpec = appearance.substring(appearance.indexOf(Appearances.EX));
+        if (exSpec.contains(")")) {
```

Comment:
I'm just wondering how helpful we should be with mistakes here. I've played around with a few examples and I don't think we end up with a crash ever (because `indexOf` returns -1 when there's no match), but we can end up with weird results. For example, `"ex:com.example().collectanswersprovider"` would give us an intent action of `"com.example"`. Maybe we should parse using something like `Pattern` (example [here](https://stackoverflow.com/questions/4662215/how-to-extract-a-substring-using-regex)) so we just get "correct" appearances. What do you think?